aws-ssm-maintenancewindow/src/main/java/software/amazon/ssm/maintenancewindow/translator/ExceptionTranslator.java

```diff
@@ -0,0 +1,82 @@
+import software.amazon.awssdk.services.ssm.model.ResourceLimitExceededException;
+import software.amazon.awssdk.services.ssm.model.FeatureNotAvailableException;
+import software.amazon.awssdk.services.ssm.model.TargetInUseException;
+import software.amazon.awssdk.services.ssm.model.IdempotentParameterMismatchException;
+import software.amazon.awssdk.services.ssm.model.TooManyUpdatesException;
+import software.amazon.awssdk.services.ssm.model.InternalServerErrorException;
+import software.amazon.awssdk.services.ssm.model.SsmRequest;
+import software.amazon.cloudformation.exceptions.BaseHandlerException;
+import software.amazon.cloudformation.exceptions.CfnAlreadyExistsException;
+import software.amazon.cloudformation.exceptions.CfnServiceLimitExceededException;
+import software.amazon.cloudformation.exceptions.CfnServiceInternalErrorException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnThrottlingException;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+
+/**
+ * Translator between service model exceptions and CloudFormation exceptions.
+ */
+public class ExceptionTranslator {
+
+    /**
+     * Translates service model exceptions to CloudFormation exceptions.
+     *
+     * @param serviceException Service model exception to translate.
+     * @param request Type of SsmRequest where the service exception came from.
+     * used for better exception messaging for the customer.
+     * @return CloudFormation-type exception converted from a service model exception.
+     */
+    public BaseHandlerException translateFromServiceException(final Exception serviceException,
+                                                              final SsmRequest request) {
+
+        if (serviceException instanceof AlreadyExistsException) {
+
+            return new CfnAlreadyExistsException(serviceException);
+        } else if (serviceException instanceof ResourceLimitExceededException) {
+
+            return new CfnServiceLimitExceededException(serviceException);
+        } else if (serviceException instanceof DoesNotExistException) {
+
+            return new CfnNotFoundException(serviceException);
+        } else if (serviceException instanceof InternalServerErrorException) {
+
+            return new CfnServiceInternalErrorException(
+                    getClassNameWithoutRequestSuffix(request.getClass().getSimpleName()),
+                    serviceException);
+        } else if (serviceException instanceof FeatureNotAvailableException
+                || serviceException instanceof TargetInUseException
+                || serviceException instanceof IdempotentParameterMismatchException){
+
+            return new CfnInvalidRequestException(request.toString(), serviceException);
+        }else if (serviceException instanceof TooManyUpdatesException) {
```

Comment:
I don't think MaintenanceWindow supports this exception. Should we include it here?

Reply:
I believe it does in context to tag expansion, we can discuss more in details offline to avoid sharing any excessive info comments

Reply:
I also do not see `TooManyUpdatesException` in the public docs for MaintenanceWindow APIs, like UpdateMaintenanceWindow: https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_UpdateMaintenanceWindow.html.

For example, it is present in UpdateAssociation: https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_UpdateAssociation.html.

If we really do expose this to customers, we should update the public docs for API documentation.

Reply:
Well make sense its not an exception directly from service so i have removed it for now. Will check on it more and see if its require will update exception handler in a follow up plus will open request to update public document.